### PR TITLE
Enable `pip install` in to a fresh virtualenv

### LINKED ISSUE #178: Requires dependencies to be installed to run setup.py
This means that installation in to a clean virtualenv isn't simple:

```
$ pip install -e .
Obtaining file:///home/daniel/dev/ubuntu-advantage-client
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/home/daniel/dev/ubuntu-advantage-client/setup.py", line 7, in <module>
        from uaclient import config
      File "/home/daniel/dev/ubuntu-advantage-client/uaclient/config.py", line 6, in <module>
        import yaml
    ModuleNotFoundError: No module named 'yaml'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /home/daniel/dev/ubuntu-advantage-client/
```

### PR DESCRIPTION
**Pull request by OddBloke**
Fixes #178.

### CODE REVIEW

**Comment by OddBloke** [comment]
I've noticed an issue with this locally; hold off on review/merge for now.

**Comment by OddBloke** [comment]
OK, now ready for review.

File: uaclient/version.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+
+def get_version(_args=None):
+    """Return the package version if set, otherwise return git describe."""
+    if not PACKAGED_VERSION.startswith('@@PACKAGED_VERSION'):
+        return PACKAGED_VERSION
+    topdir = os.path.dirname(os.path.dirname(__file__))
+    if os.path.exists(os.path.join(topdir, '.git')):
+        cmd = ['git', 'describe', '--abbrev=8', '--match=[0-9]*', '--long']
+        return check_output(cmd, text=True, encoding='utf-8').strip()
```
Comment (powersj) [inline uaclient/version.py]: Sent you a message on IRC, but the text=True is not something I am familiar with and was giving me traceback: https://paste.ubuntu.com/p/SbcmY8PGmV/

If I remove the text=True, I run into being unable to create /etc/... files, but that at least moved me on.

**Comment by powersj** [comment]
Also to follow up, I would like to see the travis tests updated to include an install and if possible a help output to verify that everything is operational from that side as well.

**Pull request merged**